CONTRIBUTING.md

```diff
@@ -1,14 +1,72 @@
 # Contributing to Bystro
+We welcome 3rd-party contributions.  Out of respect for your time, please open an issue and confirm
+the basic approach / design with the core team before submitting a PR.  Please also see our code
+[style guide](docs/style_guide.md).
+
+We use a forking model of contribution, where you fork the repository, checkout a branch in your fork, and use GitHub's UI to issue the pull request back to main repository.
+- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models#fork-and-pull-model
+- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork
+
+As an example, let's say we have forked the repository, and that fork has address `git@github.com:foobar/bystro.git`
+
+The workflow is as usual, except you use your fork as the origin
+```sh
+git clone git@github.com:foobar/bystro.git
+git checkout -b feature/some_feature
+git commit -a -m "Addresses issue #NNN"
+git push --set-upstream origin feature/some_feature
+```
+
+Then, visit your fork, and use GitHub to issue the pull request. GitHub will automatically set the bystrogenomics/master as the branch you're PR'ing against.
+- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork
+
+To keep your fork up to date with main bystro repository, in your local clone create an `upstream` remote, that points to the main repository:
+```sh
+# Create an "upstream" remote that points to the main repository
+# and then fetch the branches from the main repository
+git remote add upstream git@github.com:bystrogenomics/bystro.git
+git fetch upstream 
+```
+
+Often you may be interested in ensuring your forked repository has an up to date `master` branch:
+```sh
+git checkout master
+git pull upstream master
+```
+
+# Reviewing pull requests
+Let's say a contributor has created a pull request, and we wish to review it. That pull request has number #999.
+
+In our local clone:
+```sh
+# make a new branch, pr_999, that points to the pull request #999
+git fetch upstream pull/999/head:pr_999 
+git checkout pr_999
+```
+
+To fetch updates to the pull request (say the contributor updated their PR)
+```sh
+# assuming you're in the pr_999 branch
```

Comment:
Sorry, I don't know what you mean. This is the recommended workflow from GitHub: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally. 

Could you help me understand?

Reply:
I'm just saying: instead of writing out "make sure you're on the feature branch" we could just write: `git checkout feature-branch`, but it's not a big deal either way.